kart/point_cloud/schema_util.py

```diff
@@ -113,33 +150,26 @@ def equivalent_copc_pdrf(pdrf):
         return 6
 
 
-def pdal_schema_to_kart_schema(pdal_schema):
-    """
-    Given the JSON schema as PDAL loaded it, format it as a Kart compatiblie schema.json item.
-    Eg "type" -> "dataType", size is measured in bits.
-    """
-    return Schema(
-        [_pdal_col_schema_to_kart_col_schema(col) for col in pdal_schema["dimensions"]]
-    )
-
-
-def _pdal_col_schema_to_kart_col_schema(pdal_col_schema):
-    return {
-        "name": pdal_col_schema["name"],
-        "dataType": _pdal_type_to_kart_type(pdal_col_schema["type"]),
-        # Kart measures data-sizes in bits, PDAL in bytes.
-        "size": pdal_col_schema["size"] * 8,
-    }
-
-
-# TODO - investigate what types PDAL can actually return - it's not the same as the LAZ spec.
-# TODO - our dataset types don't have any notion of signed vs unsigned.
-_PDAL_TYPE_TO_KART_TYPE = {
-    "floating": "float",
-    "unsigned": "integer",
-    "string": "text",
+def _vlr_type_to_kart_type(vlr_datatype, options):
+    if vlr_datatype == 0:
+        return {"dataType": "blob", "length": options}
+    if vlr_datatype <= 10:
+        return _VLR_TYPE_TO_KART_TYPE[vlr_datatype]
+    array_length = int((vlr_datatype + 9) / 10)
+    vlr_datatype = (vlr_datatype - 1) % 10 + 1
```

Comment:
(a) this is confusing to reason about
(b) types >30 are actually not supported by the spec (not used as arrays) so this seems incorrect
(c) the double/triple array thing was deprecated in LAS 1.4 R14: https://github.com/ASPRSorg/LAS/issues/1

I'd suggest we drop it, YAGNI. perhaps just assert that datatype <= 10

Reply:
Good point, done.